### PR TITLE
Feature/pull from ipfs

### PR DESCRIPTION
**Pull request by anoff**
This targets #8 but is not in a fully stable phase yet.
What's your take @kevinahuber should it be merged anyhow since the docker setup is a bit beyond this issues scope?

### CODE REVIEW

**Comment by kevinahuber** [comment]
Ah this is cool. Pulling it down and checking it out now.

File: setup/docker-compose.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ services:
      - ../data/ipfs/data:/data/ipfs
      - ../data/ipfs/staging:/export
   testrpc:
-    image: "harshjv/testrpc:latest"
+    image: "testprc_persist:latest"
```
Comment (kevinahuber) [inline setup/docker-compose.yml]: Where is this? I cannot seem to find an image with this name... or `testprc` or `testrpc_persist` for that matter

**Comment by kevinahuber** [comment]
Also, you addressed #5 in the process

File: app.js [hunk 2]
```diff
@@ -6,26 +6,100 @@
...
   console.log(`${modelCount} models found`)
 
   for (let modelId = 0; modelId < modelCount; modelId++) {
     const model = await sonar.getModel(modelId)
-    console.log(`model#${modelId}: ${model.weightsAddress}`)
-    if (model.gradientCount > 0) {
-      const gradients = await sonar.getModelGradients(modelId, model.gradientCount - 1)
-      console.log(`latest gradient#${gradients.id}: ${gradients.weightsAddress[0] + gradients.weightsAddress[1]}`)
+    console.log(`model#${model.id} (${model.gradientCount}): ${model.weightsAddress}`)
+    if (model.gradientCount > Infinity) { // disable for now
```
Comment (kevinahuber) [inline app.js]: fancy way of disabling 🎲

File: setup/docker-compose.yml (hunk 1 above)
Comment (anoff) [reply setup/docker-compose.yml]: It is a local docker image that I played around with @TheOriginalAlex to get the Sonar contract already bootstrapped into `testrpc` (`prc` is a typo that I noticed to late). Was hoping we could automate the build process by now but it seems to be a tad complicated. Switching back to the previous image to have a somewhat working branch!

**Comment by anoff** [comment]
Sorrry for the rebase but it should be cleaner now 🙈

**Comment by anoff** [comment]
> Also, you addressed #5 in the process

Not necessarily, currently it's more like debug messages for myself :D Would love for someone (**first-pr** 😉 ) to review and look at it from a user perspective.

**Review by kevinahuber — APPROVED** [review]
👍 🎉

**Pull request merged**